### PR TITLE
Wrap all middleware methods for checking

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
This wraps all the middleware methods with checking. This also enables the `http` to `https` redirection on the API.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 77b981cb842f723812ee11abb6a766a085c51aa4 (NyanKiyoshi/saleor @ fix/middlewares)
Base comparison is f8d3feaf58a00548dde79fef8aae7cafb47cc8d2.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         43	         43	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

File: saleor/core/middleware.py [hunk 1]
```diff
@@ -28,25 +28,101 @@
...
 
 
-def django_only_request_handler(get_response: Callable, handler: Callable):
+class RestrictedPathRequestHandler(MiddlewareMixin):
+    """Wraps a middleware to restrict it to only perform
+    when a given condition is True."""
+
+    # The methods to wrap with the path check
+    # Refer to https://docs.djangoproject.com/en/2.2/topics/http/middleware/
+    MIDDLEWARE_METHODS = (
```
Comment (maarcingebala) [inline saleor/core/middleware.py]: `process_request` doesn't have to be included here?

Comment (NyanKiyoshi) [reply saleor/core/middleware.py]: This is actually dead code. The first idea was to dynamically wrap methods, which was why there was this list.

It was actually a bad idea for the reasons, that:
1. Too much magic
2. The methods don't all return `None`, one needs to call the chain, another needs the response returned.

Having no magic allows better flexibility and to introduce new methods if any other methods get added.

**Comment by maarcingebala** [comment]
We agreed to revert middlewares to the original state, where all of them are enabled for all requests. The code responsible for wrapping middlewares adds some complexity to the codebase, and the performance gain from it is really minor, so it may be not worth of having it.

**Pull request closed without merging**